<?xml version="1.0"?>
  <robot name="pointer_6dof" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <link name="base_link" />

    <link name="x" />

    <link name="xy" />

    <link name="xyz" />

    <link name="xyzr" />

    <link name="xyzrq" />

    <link name="xyzrqp" />

    <joint name="move_x" type="prismatic" >
      <parent link="base_link" />
      <child link="x" />
      <axis xyz="1 0 0" />
      <limit lower="0" upper="2" effort="100" velocity="100" />
    </joint>

    <joint name="move_y" type="prismatic" >
      <parent link="x" />
      <child link="xy" />
      <axis xyz="0 1 0" />
      <limit lower="0" upper="2" effort="100" velocity="100" />
    </joint>

    <joint name="move_z" type="prismatic" >
      <parent link="xy" />
      <child link="xyz" />
      <axis xyz="0 0 1" />
      <limit lower="0" upper="2" effort="100" velocity="100" />
    </joint>

    <joint name="yaw" type="revolute" >
      <parent link="xyz" />
      <child link="xyzr" />
      <axis xyz="0 0 1" />
      <limit lower="-3.141" upper="3.141" effort="100" velocity="100" />
    </joint>  

    <joint name="elev" type="revolute" >
      <parent link="xyzr" />
      <child link="xyzrq" />
      <axis xyz="0 1 0" />
      <limit lower="-3.141" upper="3.141" effort="100" velocity="100" />
    </joint>  

    <joint name="twist" type="revolute" >
      <parent link="xyzrq" />
      <child link="xyzrqp" />
      <axis xyz="1 0 0" />
      <limit lower="-3.141" upper="3.141" effort="100" velocity="100" />
    </joint>  

</robot>
